account_reconciliation_widget_due_date/static/src/js/reconciliation_model.js

```diff
@@ -0,0 +1,155 @@
+    require
+) {
+    "use strict";
+
+    var Model = require("account.ReconciliationModel");
+    var utils = require("web.utils");
+    var session = require("web.session");
+    var core = require("web.core");
+    var _t = core._t;
+    Model.StatementModel.include({
+        changeDateDue: function(handle, date_due, preserveMode) {
+            var self = this;
+            var line = this.getLine(handle);
+            line.st_line.date_due = date_due;
+            return Promise.resolve(date_due)
+                .then(function() {
+                    return self._computeLine(line);
+                })
+                .then(function() {
+                    return self.changeMode(
+                        handle,
+                        preserveMode ? line.mode : "default",
+                        true
+                    );
+                });
+        },
+        validate: function(handle) {
+            var self = this;
+            this.display_context = "validate";
+            var handles = [];
+            if (handle) {
+                handles = [handle];
+            } else {
+                _.each(this.lines, function(line, handle) {
+                    if (
+                        !line.reconciled &&
+                        line.balance &&
+                        !line.balance.amount &&
+                        line.reconciliation_proposition.length
+                    ) {
+                        handles.push(handle);
+                    }
+                });
+            }
+            var ids = [];
+            var values = [];
+            var handlesPromises = [];
+            _.each(handles, function(handle) {
+                var line = self.getLine(handle);
+                var props = _.filter(line.reconciliation_proposition, function(prop) {
+                    return !prop.invalid;
+                });
+                var computeLinePromise;
+                if (props.length === 0) {
+                    // Usability: if user has not chosen any lines and click validate, it has the same behavior
+                    // as creating a write-off of the same amount.
+                    props.push(
+                        self._formatQuickCreate(line, {
+                            account_id: [
+                                line.st_line.open_balance_account_id,
+                                self.accounts[line.st_line.open_balance_account_id],
+                            ],
+                        })
+                    );
+                    // Update balance of line otherwise it won't be to zero and another line will be added
+                    line.reconciliation_proposition.push(props[0]);
+                    computeLinePromise = self._computeLine(line);
+                }
+                ids.push(line.id);
+                handlesPromises.push(
+                    Promise.resolve(computeLinePromise).then(function() {
+                        var values_dict = {
```

Comment:
It seems you are overriding whole method, so this is not correct, as we are breaking inheritance. You just need to put the due date. On the worst case, just do an extra write.

Reply:
Sorry but I can't think of another way to do this.
I can think of several options but I think none of them are good:
- Call super () of the validate function and then redo all the code of the validate function again to send: date_due but we will have an error for trying to reconcile something that is already reconciled.
- Do not use the validate function and use the date_due value that is defined in account.bank.statement.line. This does not work because the saved value will only be the one that was previously saved when creating the records because from the reconciliation page when a date is defined the change is not saved.

Also, I think it is important to note that in 14.0 you could change this and make the dict get from a function that could be overwritten, what do you think?